src/test/java/tools/jackson/databind/ext/javatime/ser/WriteDatesAsTimestamps5142JavaTimeTests.java

```diff
@@ -0,0 +1,160 @@
+    public void testWriteDatesAsTimeStamps() throws Exception {
+        // java.time.OffsetDateTime
+        _testTimestamp(
+                LocalDateTime.of(2025, 5, 4, 18, 1, 0),
+                LocalDateTime.class,
+                // Expected "[2025,5,4,18,1,0]",
+                "[2025,5,4,18,1]", // Acutal
+                "\"2025-05-04T18:01:00\""
+        );
+        // java.time.ZonedDateTime
+        _testTimestamp(
+                LocalDateTime.of(2025, 5, 4, 18, 1, 2),
+                LocalDateTime.class,
+                "[2025,5,4,18,1,2]",
+                "\"2025-05-04T18:01:02\""
+        );
+        // java.time.LocalDate
+        _testTimestamp(
+                LocalDate.of(2025, 5, 4),
+                LocalDate.class,
+                "[2025,5,4]",
+                "\"2025-05-04\""
+        );
+        // java.time.LocalTime
+        _testTimestamp(
+                LocalTime.of(18, 1, 2),
+                LocalTime.class,
+                "[18,1,2]",
+                "\"18:01:02\""
+        );
+        // java.time.Instant
+        _testTimestamp(
+                Instant.ofEpochMilli(1234567890123L),
+                Instant.class,
+                // Expected "1234567890123",
+                "1234567890.123000000",
+                "\"2009-02-13T23:31:30.123Z\""
+        );
+        // java.time.ZoneId
+        _testTimestamp(
+                ZoneId.of("UTC"),
+                ZoneId.class,
+                "\"UTC\"",
+                "\"UTC\""
+        );
+        // java.time.ZoneOffset
+        _testTimestamp(
+                ZoneOffset.ofHours(2),
+                ZoneOffset.class,
+                "\"+02:00\"",
+                "\"+02:00\""
+        );
+        // java.time.Duration
+        _testTimestamp(
+                Duration.ofHours(2),
+                Duration.class,
+                // Expected "7200000",
+                "\"PT2H\"", // Actual
+                "\"PT2H\""
+        );
+        // java.time.Period
+        _testTimestamp(
+                Period.of(2025, 5, 4),
+                Period.class,
+                // Expected "[2025,5,4]",
+                "\"P2025Y5M4D\"", // Actual
+                "\"P2025Y5M4D\""
+        );
+        // java.time.Year
+        _testTimestamp(
+                Year.of(2025),
+                Year.class,
+                "2025", // Actual
+                "2025"
+        );
+        // java.time.YearMonth
+        _testTimestamp(
+                YearMonth.of(2025, 5),
+                YearMonth.class,
+                "[2025,5]",
+                "\"2025-05\""
+        );
+        // java.time.MonthDay
+        _testTimestamp(
+                MonthDay.of(5, 4),
+                MonthDay.class,
+                "\"--05-04\"",
+                "\"--05-04\""
+        );
+        // java.time.OffsetTime
+        _testTimestamp(
+                OffsetTime.of(18, 1, 2, 0, ZoneOffset.UTC),
+                OffsetTime.class,
+                "[18,1,2,\"Z\"]",
```

Comment:
Although Zoned variants do have distinct timestamp internally, due to nanosecond resolution cannot fit in 64-bit `long` I think? Probably why Array notation was chosen by module author.